python/src/scipp/compat/mantid.py

```diff
@@ -108,13 +111,20 @@ def ConvertEventWorkspaceToDataset(ws, load_pulse_times):
             pt = ws.getSpectrum(i).getPulseTimes()
             labs[sc.Dim.Position, i].values = np.asarray([p.total_nanoseconds()
                                                           for p in pt])
-
-    coords_and_labs = {"coords": {dim: coords, sc.Dim.Position: pos},
-                       "labels": {"spectrum_number": num,
-                                  "component_info": comp_info}}
+        w = ws.getSpectrum(i).getWeights()
+        weights[sc.Dim.Position, i].values = w
+        if not contains_weighted_events and (len(w) > 0):
+            if np.amax(w) > 1.0:
+                contains_weighted_events = True
```

Comment:
Doesn't Mantid provide a way to check whether events are weighted or nor?

Reply:
Yes, but it involves importing types like `_api.EventType.TOF` and then passing them as arguments to the `ConvertEventWorkspaceToDataset()` function.
I can do this if you prefer, and I guess finding the max could become slow for very large event lists. But since we have not taken performance into account for the other parts of this converter i thought this would be ok/less intrusive.